compiler/semexprs.nim

```diff
@@ -198,6 +198,11 @@ proc semConv(c: PContext, n: PNode): PNode =
       # separate proc from fitNode?
       if op.kind == nkSym and op.sym.isGenericRoutine:
         result.sons[1] = fitNode(c, result.typ, result.sons[1])
+      elif op.kind == nkPar and targetType.kind == tyTuple and
+                op.sons.len > 0 and op.sons[0].kind == nkExprColonExpr:
```

Comment:
You were right. Other cases, such as symbol lookup weren't handled either. fitNode seems appropriate for the heavy-lifting.